src/logger/POSTRequestLogger.ts

```diff
@@ -0,0 +1,153 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Logger, LogLevel } from '../index';
+import IntervalScheduler from '../scheduler/IntervalScheduler';
+
+export default class POSTRequestLogger implements Logger {
+  private logCapture: string[] = [];
+  private logSequenceNumber: number = 0;
+  private lock = false;
+  private static BATCH_SIZES = 75;
+  private intervalScheduler: IntervalScheduler;
+
+  constructor(
+    private name: string,
+    private level = LogLevel.WARN,
+    private meetingId: string,
+    private attendeeId: string,
+    private intervalMs: number,
+    private baseURL: string
+  ) {
+    this.intervalScheduler = new IntervalScheduler(this.intervalMs);
+    var _this = this;
+    window.addEventListener('unload', function(event) {
+        _this.intervalScheduler.stop();
+        if(!navigator.sendBeacon){
+            return
+        }
+        // Send the beacon
+        while (_this.logCapture.length > 0) {
```

Comment:
I think Beacon is only intended to be called once. Why are we splitting up the logs with multiple requests? If the limit is known to be 64KB, then can we add as many logs to the body as will fit in 64KB and make one request?

Reply:
i thought if we could break the remaining logs array into smaller chunks, we could avoid the 64kb limitation. So we will send multiple beacon calls with smaller log size.

Reply:
I think it would be better to append log lines keeping track of overall size and stop when some limit is reached. Otherwise this could potentially cause tens or hundreds of network requests which could end up being throttled by the server (which will defeat the purpose since they cannot be retried and will drop on the floor).